src/sparseml/transformers/compression/README.md

```diff
@@ -47,21 +45,22 @@ model = SparseAutoModelForCausalLM.from_pretrained(MODEL_PATH)
 sparsity_config = BitmaskConfig()
 compressor = BitmaskCompressor(config=sparsity_config)
 
+# compresses the model using Bitmask compression
 model_state_dict = model.state_dict()
 sparse_state_dict = compressor.compress(model_state_dict)
 
+# save the compressed model
+model.save_pretrained(
+    OUTPUT_PATH, 
+    safe_serialization=True, 
+    state_dict=sparse_state_dict)
 
-model.save_pretrained(OUTPUT_PATH, safe_serialization=True, state_dict=sparse_state_dict)
+# decompress the compressed state dict
+dense_state_dict = compressor.decompress(OUTPUT_PATH)
 
-safetensors_path = os.path.join(OUTPUT_PATH, "model-00001-of-00002.safetensors")
-with safe_open(safetensors_path, framework="pt", device=0) as f:
-    test_name = "model.layers.4.self_attn.k_proj.weight"
-    bitmask = f.get_tensor(test_name + ".bitmask")
-    shape = f.get_tensor(test_name + ".shape")
-    values = f.get_tensor(test_name + ".compressed")
-    row_offsets = f.get_tensor(test_name + ".row_offsets")
-    print(f"bitmask: {bitmask}")
-    print(f"shape: {shape}")
-    print(f"values: {values}")
-    print(f"row offsets: {row_offsets}")
+# use the dense state dict to reload the model
+model_again = model = SparseAutoModelForCausalLM.from_pretrained(
+    OUTPUT_PATH, 
+    state_dict=dense_state_dict
+)
```

Comment:
1. Is `SparseAutoModelForCausalLM` required here? it seems like AutoModelForCausalLM would work as well since the state_dict has already been materialized
2. Will this temporarily double the memory usage since the weights are fully materialized in dense_state_dict and AutoModelForCausalLM will create an empty model before using state_dict?

Reply:
1. Yes AutoModelForCausalLM would work here as well
2. I didn't think passing in the state dict would create a second copy of the model, but will check this using the method you posted above

Reply:
@mgoin made some updates to the decompress function to return an iterator, so we can now load the model then insert the decompressed layers one by one. The updated example includes peak memory usage. I didn't add anything on memory usage into the unit tests because GHA runs on cpu only